packages/sdk/src/reliability_monitor/receiver.ts

```diff
@@ -49,6 +51,13 @@ export class ReceiverReliabilityMonitor {
       }
     };
     allPeerIdStr.forEach((peerId) => this.missedMessagesByPeer.set(peerId, 0));
+
+    this.addLibp2pEventListener("peer:disconnect", (evt) => {
+      const peerId = evt.detail;
+      if (this.getPeers().some((p) => p.id.equals(peerId))) {
```

Comment:
one last thing. do you thing we can try to use metadata here to identify peers that were used for Filter? 
I am afraid of re-subscribing to peers that were not used for subscription before (tho seems not to be the case, at least I didn't hit it yet)

Reply:
part of https://github.com/waku-org/js-waku/issues/2167